stellargraph/data/converter.py

```diff
@@ -147,19 +147,16 @@ def add_all_attributes(
             raise TypeError(
                 "Converter should be a subclass of StellarAttributeConverter"
             )
-        if not isinstance(graph, StellarGraphBase):
-            raise TypeError("Graph should be a StellarGraph")
+        if not isinstance(graph, StellarGraph):
+            raise TypeError("Graph must be a StellarGraph or StellarDigraph object")
 
         # Go through graph to find node attributes
-        all_attrs = set(
-            k for v in graph.nodes_of_type(node_type) for k in graph.nodes[v].keys()
-        )
+        attrs = set()
+        for node in graph.nodes_of_type(node_type):
+            attrs = attrs | graph.node_attributes(node)
```

Comment:
Agreed - lots of this code I didn't write, or simply worked around with minimal changes. I think I rewrite it a bit in the subsequent branch.

Reply:
Hmm, I will look at it to remind myself what I did and why.

Reply:
Ok, direct mutation is more sensible. The comprehension was removed because the encapsulation now returns a list of attributes, hiding the implementation.